### PR TITLE
TYP: nattype.pyi

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/_libs/tslibs/nattype.pyi [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+nat_strings: set[str]
+
+def is_null_datetimelike(val: object, inat_is_null: bool = ...) -> bool: ...
+
+class NaTType(datetime):
+    value: np.int64
+
+    def asm8(self) -> np.datetime64: ...
+    def to_datetime64(self) -> np.datetime64: ...
+    def to_numpy(self, dtype=..., copy: bool = False) -> np.datetime64: ...
```
Comment (simonjayhawkins) [inline pandas/_libs/tslibs/nattype.pyi:17]:
```suggestion
    def to_numpy(self, dtype=..., copy: bool = ...) -> np.datetime64: ...
```

File: pandas/io/excel/_odfreader.py [hunk 2]
```diff
@@ -202,7 +202,8 @@ def _get_cell_value(self, cell, convert_float: bool) -> Scalar:
         elif cell_type == "time":
             result = pd.to_datetime(str(cell))
             result = cast(pd.Timestamp, result)
-            return result.time()
+            # error: Item "str" of "Union[float, str, NaTType]" has no attribute "time"
+            return result.time()  # type: ignore[union-attr]
```
Comment (simonjayhawkins) [inline pandas/io/excel/_odfreader.py:206]: observation. `pd.Timestamp` resolves to `Any`, so cast above has no effect.

File: pandas/io/formats/format.py [hunk 3]
```diff
@@ -1796,17 +1799,13 @@ def get_format_timedelta64(
 
     If box, then show the return in quotes
     """
-    values_int = values.view(np.int64)
+    values_int = np.asarray(values.view(np.int64))
```
Comment (simonjayhawkins) [inline pandas/io/formats/format.py:1802]: I think you shouldn't need to do this. once the overloads for view are more comprehensive, mypy will know that `.view(np.int64)` returns an numpy array.

I would leave this unchanged for now.

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @jbrockmendel. generally lgtm. just the code change for mypy is a concern. `TimedeltaIndex.view(np.int64)` and `TimedeltaArray.view(np.int64)` both return a numpy array.

File: pandas/io/excel/_odfreader.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/io/excel/_odfreader.py:206]: yah, Timestamp is pretty much going to be left for last in this process.  Among other things, mypy complains about `__new__` sometimes returning NaT.

**Comment by jbrockmendel** [comment]
updated  + green

**Review by simonjayhawkins — APPROVED** [review]
Thanks @jbrockmendel for the updates. have merged master to be sure.

File: pandas/core/internals/construction.py [hunk 4]
```diff
@@ -368,7 +368,10 @@ def maybe_squeeze_dt64tz(dta: ArrayLike) -> ArrayLike:
     # TODO(EA2D): kludge not needed with 2D EAs
     if isinstance(dta, DatetimeArray) and dta.ndim == 2 and dta.tz is not None:
         assert dta.shape[0] == 1
-        dta = dta[0]
+        # error: Incompatible types in assignment (expression has type
+        # "Union[DatetimeLikeArrayMixin, Union[Any, NaTType]]", variable has
+        # type "Union[ExtensionArray, ndarray]")
+        dta = dta[0]  # type: ignore[assignment]
```
Comment (jreback) [inline pandas/core/internals/construction.py:374]: if you return you can avoid this i think

**Review by jreback — APPROVED** [review]
minor comment for followup

**Pull request merged**